editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -3211,10 +3240,49 @@ void CanvasItemEditor::_draw_selection() {
 				}
 			}
 
-			// Draw the rescale handles
+			// Draw the move handles
 			bool is_ctrl = Input::get_singleton()->is_key_pressed(KEY_CONTROL);
 			bool is_alt = Input::get_singleton()->is_key_pressed(KEY_ALT);
-			if ((is_alt && is_ctrl) || tool == TOOL_SCALE || drag_type == DRAG_SCALE_X || drag_type == DRAG_SCALE_Y) {
+			if (tool == TOOL_MOVE && show_transformation_gizmos) {
+				if (_is_node_movable(canvas_item)) {
+					Transform2D unscaled_transform = (xform * canvas_item->get_transform().affine_inverse() * canvas_item->_edit_get_transform()).orthonormalized();
+					Transform2D simple_xform = viewport->get_transform() * unscaled_transform;
+
+					Size2 move_factor = Size2(MOVE_HANDLE_DISTANCE, MOVE_HANDLE_DISTANCE);
+					viewport->draw_set_transform_matrix(simple_xform);
+
+					Vector<Point2> points;
+					points.push_back(Vector2(move_factor.x * EDSCALE, 5 * EDSCALE));
+					points.push_back(Vector2(move_factor.x * EDSCALE, -5 * EDSCALE));
+					points.push_back(Vector2((move_factor.x + 10) * EDSCALE, 0));
+
+					Vector<Color> colors;
```

Comment:
Couldn't you use `draw_colored_polygon` instead of pushing 3 colors every time ?

Reply:
Implemeted. Yes that's defiantly the better way